pepper-apis/scripts/dsm/tests/pecgs-phase4-participant.sh

```diff
@@ -0,0 +1,624 @@
+     -H "cookie: _ga=GA1.2.662503047.1684868999; _ga_KB65C93K2D=GS1.1.1687527731.2.0.1687527731.0.0.0" \
+     -H "referer: $dsm_url/$study/participantList" \
+     -H "sec-ch-ua: \"Not.A/Brand\";v=\"8\", \"Chromium\";v=\"114\", \"Google Chrome\";v=\"114\"" \
+     -H "sec-ch-ua-mobile: ?0" \
+     -H "sec-ch-ua-platform: \"macOS\"" \
+     -H "sec-fetch-dest: empty" \
+     -H "sec-fetch-mode: cors" \
+     -H "sec-fetch-site: same-origin" \
+     -H "user-agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36" \
+     --compressed )
+
+   echo $response
+
+  existing_order=$(echo "$response" | jq '.[] | select(.sampleType=="Tumor" and .lastOrderNumber != null)')
+
+
+  if [[ $existing_order != "" ]]; then
+    echo "The existing order is: $existing_order"
+
+    pdoOrderId=$(echo $existing_order | jq -r '.pdoOrderId')
+
+    if [[ pdoOrderId != "" ]];then
+      echo "PDO already exists"
+      exit
+    fi
+    lastOrderNumber=$(echo $existing_order | jq -r '.lastOrderNumber')
+
+    echo "order number is $lastOrderNumber"
+
+    project="broad-ddp-$env"
+
+    pdo="PDO-$(openssl rand -base64 32 | tr -dc "a-zA-Z0-9" | head -c6)"
+
+    echo "PDO number is $pdo"
+
+    topic="$env-mercury-dsm-task"
+
+    gcloud --project=$project pubsub topics publish $topic \
+    --message "{\"status\":{\"orderID\":\"$lastOrderNumber\",\"orderStatus\":\"Approved\",\"details\":\"Successfully created order\",\"pdoKey\": \"$pdo\",\"json\": \"\"}}"
+
+    echo "published status message to pubsub"
+  else
+    if ! send-and-receive-saliva; then
+        echo "Participant does not have a saliva sample"
+        exit -1
+    fi
+
+    if ! set-onc-history-values; then
+      echo "Could not set onc history values"
+      exit -1
+    fi
+  fi
+
+}
+
+function search-received-kits {
+
+  echo "Searching received SALIVA kits for $guid"
+  
+  response=$(curl "$dsm_url/ui/kitRequests?realm=$study&target=received&kitType=SALIVA" \
+               -H "authority: $dsm_url" \
+               -H "accept: application/json" \
+               -H "accept-language: en-US,en;q=0.9" \
+               -H "authorization: Bearer $auth0Token" \
+               -H "content-type: application/json" \
+               -H "cookie: _ga=GA1.2.662503047.1684868999; _ga_KB65C93K2D=GS1.1.1687527731.2.0.1687527731.0.0.0" \
+               -H "referer: $dsm_url/$study/shippingReceived" \
+               -H "sec-ch-ua: \"Not.A/Brand\";v=\"8\", \"Chromium\";v=\"114\", \"Google Chrome\";v=\"114\"" \
+               -H "sec-ch-ua-mobile: ?0" \
+               -H "sec-ch-ua-platform: \"macOS\"" \
+               -H "sec-fetch-dest: empty" \
+               -H "sec-fetch-mode: cors" \
+               -H "sec-fetch-site: same-origin" \
+               -H "user-agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36" \
+               --compressed)
+
```

Comment:
Do you think it would be helpful to echo the response here?

Reply:
It's too long, all the received kits for that study are here